feat(doc): Mention unsafe_cross_namespace_identity in documentation (#3292)
openbao/openbao: website/content/docs/concepts/namespaces/index.mdx

namespace. Parent namespaces can also **assert** policies on identities within
a child namespace.

### Cross-namespaces identities

> Entities or groups from a parent namespace are not enabled by default.

Reviewer: ```suggestion
Entities or groups from any namespace are not enabled by default.
```

> The configuration setting `unsafe_cross_namespace_identity` gives compatibility with Vault Enterprise's cross-namespace group membership and should be enabled.

Reviewer: ```suggestion
The configuration setting `unsafe_cross_namespace_identity` gives compatibility with Vault Enterprise's cross-namespace group membership and should not be enabled. There are no mitigating controls on what identities or groups another namespace can reference and allows for unconfined permission sharing.
```

Reviewer: Preferably a `:::warning` block should be used. We may also add a note to this in the future as new features like namespace sealing will not be compatible with this unsafe behavior.

Example:

```hcl
api_addr = "..."
listener "tcp" {
  ...
}

> # Allow child namespaces to reference parent namespaces entities or groups

Reviewer: ```suggestion
# Allow namespaces to reference other namespaces' entities or groups
```

namespace. Parent namespaces can also **assert** policies on identities within
a child namespace.

### Cross-namespaces identities

Entities or groups from any namespace are not enabled by default.

:::warning

The configuration setting `unsafe_cross_namespace_identity` gives compatibility with Vault Enterprise's cross-namespace group membership and should not be enabled.

> **Warning**: There are no mitigating controls on what identities or groups another namespace can reference and allows for unconfined permission sharing.

Reviewer 2: I believe the last part of the sentence is missing some words? _What_ allows for unconfined permission sharing?

Reviewer: @satoqz I kinda feel it is the earlier (the lack of mitigating controls) allows for unconfined permission sharing. I think in some places you'd maybe state `and thus`, but I think it is fine either way?

> 

Reviewer 2: ```suggestion
There are no mitigating controls on what identities or groups another namespace can reference and allows for unconfined permission sharing.

:::

```

No need for a warning inside of a warning, but the block also needs to be closed.
